measurements.py

```diff
@@ -220,9 +338,24 @@ class TimeMeasurer(AnalogMeasurer):
         measures=[positive_width, negative_width, positive_duty, negative_duty])
 
     def measure_range(self, data: AnalogSpan):
+        if data.clipped:
+            self.positive_width.error = CLIPPED_ERROR
+            self.negative_width.error = CLIPPED_ERROR
+            self.positive_duty.error = CLIPPED_ERROR
+            self.negative_duty.error = CLIPPED_ERROR
+            self.cycle_rms.error = CLIPPED_ERROR
+
         start_index = int(len(data) / 2)
 
         mid = (data.max + data.min) / 2
+        hysteresis = (data.max - data.min) * FREQUENCY_HYSTERESIS_PCT
+
+        start_value = data[start_index]
+        if (abs(start_value - mid) < hysteresis):
+            lt_index = data.find_lt(mid - hysteresis, start=start_index)
```

Comment:
what do you think  about checking right and left, above and below, and taking the minimum of the 4?

Reply:
TODO, we will revisit this in a follow-up